app-backend/db/src/main/scala/AnnotationDao.scala

```diff
@@ -364,4 +364,144 @@ object AnnotationDao extends Dao[Annotation] {
       )
     }
   }
+
+  def listDetectionLayerAnnotationsByTaskStatus(
+      projectId: UUID,
+      layerId: UUID,
+      taskStatuses: List[String]
+  ): ConnectionIO[List[Annotation]] = {
+    val taskStatusF: List[Option[Fragment]] = taskStatuses.map(taskStatus => {
+      Some(fr"tasks.status = ${taskStatus}::task_status")
+    })
+    (fr"""
+      SELECT 
+        annotations.id,
+        annotations.project_id,
+        annotations.created_at,
+        annotations.created_by,
+        annotations.modified_at,
+        annotations.owner,
+        project_labels.name,
+        annotations.description,
+        annotations.machine_generated,
+        annotations.confidence,
+        annotations.quality,
+        annotations.geometry,
+        annotations.annotation_group,
+        annotations.labeled_by,
+        annotations.verified_by,
+        annotations.project_layer_id,
+        annotations.task_id
+      FROM annotations
+      JOIN tasks
+      ON ST_Intersects(tasks.geometry, annotations.geometry)
+      JOIN (
+        SELECT labels.id, labels.name
+        FROM
+            projects,
+            jsonb_to_recordset((projects.extras->'annotate')::jsonb ->'labels') AS labels(id uuid, name text)
+        WHERE projects.id = ${projectId}
+      ) project_labels
+      ON project_labels.id::uuid = annotations.label::uuid
+      WHERE
+        tasks.project_id = ${projectId}
+        AND tasks.project_layer_id = ${layerId}
+        AND annotations.project_id = ${projectId}
+        AND annotations.project_layer_id = ${layerId}
+        AND (""" ++ Fragments.orOpt(taskStatusF: _*) ++ fr")")
```

Comment:
Something seems non-conventional here with all the fragment quoting. Is a task status always required? Then doing something like `Fragments.in` with a `NonEmptyList` of `TaskStatus` would be clearer to future users I think.